examples/nbconverted/training_with_logging_example.py

```diff
@@ -0,0 +1,288 @@
+    raise FileNotFoundError(f"No dataset.json found in {data_dir}.")
+# dataset_index_json = dataset_index_json[0]
+
+with open(dataset_index_json, "r") as f:
+    dataset_index = json.load(f)
+
+index = pd.DataFrame(dataset_index['file_index']['records'])
+print(f"Total number of samples: {len(index)}")
+index = index.sample(n=40, random_state=42).reset_index(drop=True)
+print(f"Subset number of samples: {len(index)}")
+print(index.head())
+
+
+# ## Define custom dataset that simply crops the center patch and does normalization
+
+# In[3]:
+
+
+class SimpleDataset(Dataset):
+    def __init__(self, index_df, data_dir):
+        self.index_df = index_df
+        self.data_dir = data_dir
+
+    def __len__(self):
+        return len(self.index_df)
+
+    def __getitem__(self, idx):
+        row = self.index_df.iloc[idx]
+
+        # Load images
+        brightfield_path = self.data_dir / 'data' / row['OrigBrightfield']
+        dna_path = self.data_dir / 'data' / row['OrigDNA']
+
+        brightfield_img = Image.open(brightfield_path)
+        brightfield_img.convert('I;16')
+        dna_img = Image.open(dna_path)
+        dna_img.convert('I;16')
+        brightfield_img = np.asarray(brightfield_img, dtype=np.float32)
+        dna_img = np.asarray(dna_img, dtype=np.float32)
+
+        # Get image dimensions
+        width, height = brightfield_img.shape
+
+        # Calculate crop coordinates for center 256x256
+        left = (width - 256) // 2
+        top = (height - 256) // 2
+        right = left + 256
+        bottom = top + 256
+
+        # Crop center 256x256
+        brightfield_crop = brightfield_img[top:bottom,left:right]
+        dna_crop = dna_img[top:bottom,left:right]
+
+        # Normalize by 16bit max value
+        factor = 2 ** 16 - 1
+        brightfield_crop = brightfield_crop / factor
+        dna_crop = dna_crop / factor
+
+        # Convert to tensor
+        brightfield_tensor = torch.from_numpy(brightfield_crop).unsqueeze(0)  # Add channel dimension
+        dna_tensor = torch.from_numpy(dna_crop).unsqueeze(0)  # Add channel dimension
+
+        return brightfield_tensor, dna_tensor
+
+# Create dataset instance
+dataset = SimpleDataset(index, data_dir)
+print(f"Dataset created with {len(dataset)} samples")
+
+
+# In[4]:
+
+
+# Plot the first 5 samples from the dataset
+fig, axes = plt.subplots(5, 2, figsize=(8, 16))
+
+for i in range(5):
+    brightfield, dna = dataset[i]
+    brightfield = brightfield.numpy().squeeze()
+    dna = dna.numpy().squeeze()
+
+    # Plot brightfield image
+    axes[i, 0].imshow(brightfield.squeeze(), cmap='gray')
+    axes[i, 0].set_title(f'Sample {i} - Brightfield')
+    axes[i, 0].axis('off')
+
+    # Plot DNA image
+    axes[i, 1].imshow(dna.squeeze(), cmap='gray')
+    axes[i, 1].set_title(f'Sample {i} - DNA')
+    axes[i, 1].axis('off')
+
+plt.tight_layout()
+plt.show()
+
+
+# ## Configure and train
+
+# In[ ]:
+
+
+# Hyperparameters
+batch_size = 16
+epochs = 100
```

Comment:
Can you add any documentation here for why you have chosen these parameters similar to the learning rate? Might be helpful for not just the reader but also yourself for the future.